runtime/cudaq/qis/qkernel_ref.h

```diff
@@ -0,0 +1,195 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#pragma once
+
+#include <functional>
+#include <iostream>
```

Comment:
We should drop the `<iostream>` include, but we will need to add `<memory>` for the `unique_ptr` usage.